src/emqttd_auth_clientid.erl

```diff
@@ -40,87 +30,71 @@
-%% @end
-%%------------------------------------------------------------------------------
 -spec add_clientid(binary()) -> {atomic, ok} | {aborted, any()}.
 add_clientid(ClientId) when is_binary(ClientId) ->
     R = #mqtt_auth_clientid{client_id = ClientId},
     mnesia:transaction(fun mnesia:write/1, [R]).
 
-%%------------------------------------------------------------------------------
 %% @doc Add clientid with password
-%% @end
-%%------------------------------------------------------------------------------
 -spec add_clientid(binary(), binary()) -> {atomic, ok} | {aborted, any()}.
 add_clientid(ClientId, Password) ->
     R = #mqtt_auth_clientid{client_id = ClientId, password = Password},
     mnesia:transaction(fun mnesia:write/1, [R]).
 
-%%------------------------------------------------------------------------------
 %% @doc Lookup clientid
-%% @end
-%%------------------------------------------------------------------------------
--spec lookup_clientid(binary()) -> list().
+-spec lookup_clientid(binary()) -> list(#mqtt_auth_clientid{}).
 lookup_clientid(ClientId) ->
     mnesia:dirty_read(?AUTH_CLIENTID_TAB, ClientId).
 
-%%------------------------------------------------------------------------------
 %% @doc Lookup all clientids
-%% @end
-%%------------------------------------------------------------------------------
 -spec all_clientids() -> list(binary()).
-all_clientids() ->
-    mnesia:dirty_all_keys(?AUTH_CLIENTID_TAB).
+all_clientids() -> mnesia:dirty_all_keys(?AUTH_CLIENTID_TAB).
 
-%%------------------------------------------------------------------------------
 %% @doc Remove clientid
-%% @end
-%%------------------------------------------------------------------------------
 -spec remove_clientid(binary()) -> {atomic, ok} | {aborted, any()}.
 remove_clientid(ClientId) ->
     mnesia:transaction(fun mnesia:delete/1, [{?AUTH_CLIENTID_TAB, ClientId}]).
 
-%%%=============================================================================
-%%% emqttd_auth_mod callbacks
-%%%=============================================================================
+%%--------------------------------------------------------------------
+%% emqttd_auth_mod callbacks
+%%--------------------------------------------------------------------
 
 init(Opts) ->
     mnesia:create_table(?AUTH_CLIENTID_TAB, [
             {ram_copies, [node()]},
             {attributes, record_info(fields, ?AUTH_CLIENTID_TAB)}]),
     mnesia:add_table_copy(?AUTH_CLIENTID_TAB, node(), ram_copies),
-    case proplists:get_value(file, Opts) of
-        undefined -> ok;
-        File      -> load(File)
-    end,
+    load(proplists:get_value(file, Opts)),
     {ok, Opts}.
 
-check(#mqtt_client{client_id = undefined}, _Password, []) ->
-    {error, "ClientId undefined"};
+check(#mqtt_client{client_id = undefined}, _Password, _Opts) ->
+    {error, clientid_undefined};
 check(#mqtt_client{client_id = ClientId, peername = {IpAddress, _}}, _Password, []) ->
```

Comment:
Line 77 is too long: check(#mqtt_client{client_id = ClientId, peername = {IpAddress, _}}, _Password, []) ->.